extensions/apps/extensions/src/components/pages/extension-edit-page/contributors-page.tsx

```diff
@@ -0,0 +1,305 @@
+    } catch (error) {
+      showNotification(NotificationTypes.Error, error);
+    }
+  }, [extensionId, showNotification]);
+
+  // fetch the draft extensions that are saved only on local storage
+  const draftExtensions: Extension[] = useMemo(() => {
+    try {
+      return JSON.parse(localStorage.getItem(`${DRAFT_EXTENSIONS}-${authenticatedDID}`)) || [];
+    } catch (error) {
+      showNotification(NotificationTypes.Error, error);
+    }
+  }, [authenticatedDID, showNotification]);
+
+  const extensionData = draftExtensions.find(draftExtension => draftExtension.id === extensionId);
+
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const autoCompleteRef = useCloseActions(() => {
+    setShowSuggestions(false);
+  });
+
+  const [searchValue, setSearchValue] = useState('');
+  const [addedContributors, setAddedContributors] = useState([]);
+
+  const defaultContributorsDIDs = useMemo(() => {
+    return formValue.lastCompletedStep > 2 ? formValue?.contributors : extensionData?.contributors;
+  }, [extensionData, formValue]);
+
+  // fetch the contributors profiles
+  const { profilesData, loading, error } = useProfilesList(defaultContributorsDIDs);
+
+  // hydrate the list of added contributors with profiles
+  useEffect(() => {
+    if (profilesData.length > 0) {
+      setAddedContributors(profilesData);
+    }
+  }, [profilesData]);
+
+  // contributors are a set of profiles for which the user is following and the profile is also
+  // following the user, same as for mentions, that's why the same hook can be used
+  const { setMentionQuery, mentions: contributors } = useMentions(authenticatedDID);
+  const handleGetContributors = (query: string) => {
+    setMentionQuery(query);
+  };
+
+  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
+    const query = ev.currentTarget.value;
+    setSearchValue(query);
+    handleGetContributors(query);
+    setShowSuggestions(true);
+  };
+
+  const handleAddContributor = (profile: AkashaProfile) => {
+    if (addedContributors?.length < MAX_CONTRIBUTORS) {
+      setAddedContributors(prev => {
+        const res = Array.from(new Set([...prev, profile]));
+        return res;
+      });
+      setSearchValue('');
+      setShowSuggestions(false);
+    }
+  };
+
+  const handleRemoveContributor = (profile: AkashaProfile) => {
+    setAddedContributors(prev => prev.filter(addedProfile => profile.id !== addedProfile?.id));
+  };
+
+  const handleNavigateBack = () => {
+    navigate({
+      to: '/edit-extension/$extensionId/step3',
+    });
+  };
+
+  const [, setForm] = useAtom<FormData>(useContext(AtomContext));
+
+  const handleSave = () => {
+    setForm(prev => {
+      return {
+        ...prev,
+        // save only the DIDs of the contributors
+        contributors: addedContributors.map(contribProfile => contribProfile?.did?.id),
+      };
+    });
+    navigate({
+      to: '/edit-extension/$extensionId/step3',
+    });
+  };
+
+  return (
```

Comment:
Can we handle not authenticated state ?

Reply:
It's handled on the parent route for the extension form,  main-page.tsx. This page is a child of it.